### PR TITLE
Fix tests branch

### PR DESCRIPTION
**Pull request by karthikprabhuvinod**
Changes proposed in this pull request:
- Fix the network test to use Network Manager instead of systemd-network
- Fix the Storage test which fails sgdisk while trying to write to a partition table
- Fix the telemetry test which needed a telemetry spool directory
- Also make builds fail when one of the tests fail

### CODE REVIEW

**Comment by karthikprabhuvinod** [comment]
@mdhorn : this is what I was talking abt:

https://github.com/clearlinux/clr-installer/blob/master/storage/storage_test.go#L552-L584

**Comment by karthikprabhuvinod** [comment]
@mdhorn Looks like the sgdisk was failing as the partition creation was being skipped as those blk devices never had make partition set to true being tested for in ops.go line 453 to 455.

I have marked some part of the test while relate to advanced partitioning as obselete as i dont understand why they need to be there. Do you know?

I will wait for you to answer..

File: storage/storage_test.go [hunk 1]
```diff
@@ -590,10 +590,10 @@ func TestWritePartition(t *testing.T) {
 			}
 		}()
 		bd.Name = path.Base(file)
-		part1 := &BlockDevice{Name: bd.Name + "p1", FsType: "vfat", Size: 157286400, PartitionLabel: "CLR_BOOT", Type: BlockDeviceTypePart, MountPoint: "/boot"}
-		part2 := &BlockDevice{Name: bd.Name + "p2", FsType: "swap", Size: 125829120, PartitionLabel: "CLR_SWAP", Type: BlockDeviceTypePart, MountPoint: ""}
-		part3 := &BlockDevice{Name: bd.Name + "p3", FsType: "ext4", Size: 502267904, PartitionLabel: "CLR_ROOT_F_E", Type: BlockDeviceTypePart, MountPoint: "/"}
-		part4 := &BlockDevice{Name: bd.Name + "p4", FsType: "ext4", Size: 502267904, PartitionLabel: "CLR_MNT_/home", Type: BlockDeviceTypeCrypt, MountPoint: "/home"}
+		part1 := &BlockDevice{Name: bd.Name + "p1", FsType: "vfat", Size: 157286400, PartitionLabel: "CLR_BOOT", Type: BlockDeviceTypePart, MountPoint: "/boot", MakePartition: true}
```
Comment (mdhorn) [inline storage/storage_test.go]: Good catch, this is why the partiton were not created.

File: storage/storage_test.go [hunk 2]
```diff
@@ -590,10 +590,10 @@ func TestWritePartition(t *testing.T) {
...
+		part2 := &BlockDevice{Name: bd.Name + "p2", FsType: "swap", Size: 125829120, PartitionLabel: "CLR_SWAP", Type: BlockDeviceTypePart, MountPoint: "", MakePartition: true}
+		part3 := &BlockDevice{Name: bd.Name + "p3", FsType: "ext4", Size: 502267904, PartitionLabel: "CLR_ROOT_F_E", Type: BlockDeviceTypePart, MountPoint: "/", MakePartition: true}
```
Comment (mdhorn) [inline storage/storage_test.go]: We should drop the _E here as we do not currently allow encryption during advanced.
```suggestion
		part3 := &BlockDevice{Name: bd.Name + "p3", FsType: "ext4", Size: 502267904, PartitionLabel: "CLR_ROOT_F", Type: BlockDeviceTypePart, MountPoint: "/", MakePartition: true}
```

File: storage/storage_test.go [hunk 3]
```diff
@@ -631,15 +631,17 @@ func TestWritePartition(t *testing.T) {
...
-		if len(found) > 0 {
-			t.Fatalf("Should not have found any advanced targets %+v", found)
-		}
-
-		if !AdvancedPartitionsRequireEncryption(bds) {
-			t.Fatalf("Advanced targets should require encryption")
-		}
+		/*
+			found := FindAdvancedInstallTargets(bds)
+			if len(found) > 0 {
```
Comment (mdhorn) [inline storage/storage_test.go]: We should leave these checks, but the logic now needs to be reversed.
```suggestion
			if len(found) == 0 {
```

File: storage/storage_test.go [hunk 4]
```diff
@@ -631,15 +631,17 @@ func TestWritePartition(t *testing.T) {
...
+				t.Fatalf("Should not have found any advanced targets %+v", found)
```
Comment (mdhorn) [inline storage/storage_test.go]:
```suggestion
				t.Fatalf("Should have found any advanced targets %+v", found)
```

File: storage/storage_test.go [hunk 5]
```diff
@@ -631,15 +631,17 @@ func TestWritePartition(t *testing.T) {
...
+			}
+		*/
+		/*
+			if !AdvancedPartitionsRequireEncryption(bds) {
```
Comment (mdhorn) [inline storage/storage_test.go]:
```suggestion
			if AdvancedPartitionsRequireEncryption(bds) {
```

File: storage/storage_test.go [hunk 6]
```diff
@@ -631,15 +631,17 @@ func TestWritePartition(t *testing.T) {
...
+				t.Fatalf("Advanced targets should require encryption")
```
Comment (mdhorn) [inline storage/storage_test.go]:
```suggestion
				t.Fatalf("Advanced targets should not enable encryption")
```

File: network/network_test.go [hunk 7]
```diff
@@ -208,7 +208,7 @@ func TestApply(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	etcDir := filepath.Join(dir, systemdNetworkdDir)
+	etcDir := filepath.Join(dir, networkManagerDir)
```
[the remark this answers is not in the document]
Comment (karthikprabhuvinod) [reply network/network_test.go]: Thanks Done

File: storage/storage_test.go (hunk 2 above)
Comment (karthikprabhuvinod) [reply storage/storage_test.go]: Done

File: storage/storage_test.go (hunk 3 above)
Comment (karthikprabhuvinod) [reply storage/storage_test.go]: Done

File: storage/storage_test.go (hunk 6 above)
Comment (karthikprabhuvinod) [reply storage/storage_test.go]: Done

File: storage/storage_test.go (hunk 5 above)
Comment (karthikprabhuvinod) [reply storage/storage_test.go]: Done

File: storage/storage_test.go (hunk 4 above)
Comment (karthikprabhuvinod) [reply storage/storage_test.go]: Done

**Comment by karthikprabhuvinod** [comment]
@mdhorn Addressed all concerns..Thanks for those remarks especially to keep advanced partition checkk. Please take a relook

**Comment by karthikprabhuvinod** [comment]
@mdhorn I added the fix ...now those tests are passing..but a telmetry test i failing as it doesnt have a /var/pool/telemetry directory to copy records to. Any idea what needs to be done for this?

**Comment by karthikprabhuvinod** [comment]
i fixed the spool by creating a directory during the test

File: telemetry/telemetry_test.go [hunk 8]
```diff
@@ -361,6 +361,14 @@ func TestCopyRecords(t *testing.T) {
 		_ = os.RemoveAll(dir)
 	}()
 
+	_, err = os.Stat(telemetrySpoolDir)
```
Comment (mdhorn) [inline telemetry/telemetry_test.go]: If you are creating the directory, then you need to clean it up as well.
NOTE: The `TestLogRecord` test below also expects the directory to exist so will need a similar or shared fix.

The other approach I considered was skipping the test if the spool directory is missing:

```
diff --git a/telemetry/telemetry_test.go b/telemetry/telemetry_test.go
index f4da46aacbe3..65714267d7d9 100644
--- a/telemetry/telemetry_test.go
+++ b/telemetry/telemetry_test.go
@@ -347,6 +347,10 @@ func TestCopyRecords(t *testing.T) {
                t.Skip("Not running as 'root', skipping record copy test")
        }

+       if exists, err := utils.FileExists(telemetrySpoolDir); err == nil && !exists {
+               t.Skip("Not running due to missing telemetry bundle, skipping record copy test")
+       }
+
        url = "http://www.google.com"
        if err := telem.SetTelemetryServer(url, tid, policy); err != nil {
                t.Fatalf("Setting telemetry server (%q) should not return error: %s\n", url, err)
@@ -387,6 +391,10 @@ func TestLogRecord(t *testing.T) {
        if !utils.IsRoot() {
                t.Skip("Not running as 'root', skipping log record test")
        }
+       if exists, err := utils.FileExists(telemetrySpoolDir); err == nil && !exists {
+               t.Skip("Not running due to missing telemetry bundle, skipping log record test")
+       }
+
        if err := telem.LogRecord("success", 2, "Hello"); err != nil {
                t.Logf("TestLogRecord failed with %s", err)
        }
```

But you get more coverage with your solution.

Comment (karthikprabhuvinod) [reply telemetry/telemetry_test.go]: Okay , I think I will clean up the directory with a defer as the code gets tested

**Pull request merged**